libs/wingc/src/type_check.rs

```diff
@@ -545,7 +545,7 @@ impl Subtype for Type {
 			}
 			(Self::Function(l0), Self::Interface(r0)) => {
 				// TODO: Hack to make functions compatible with interfaces
-				// Remove this after https://github.com/winglang/wing/issues/1448
+				// Remove this after https://github.com/winglang/wing/issues/1448 // Closed
```

Comment:
There's a feature in wing where a preflight class instance that has an `inflight` `handle` method can be used in place of a inflight function if they both have the same signature. Something like
```
q.setConsumer(inflight (m: str) => {
  log("message ${m} consumed");
});
```
Is in the same as
```
class MyConsumer {
  inflight handle(m: str): void {
    log("message ${m} consumed");    
  }
}
let x = new MyConsumer();
q.setConsumer(x);
```
Now for the type checker to work it needs to be aware of this rule. I'm not going to go into _why_ we have this rule, it's related to something we call _lifting_, and perhaps it's worth another explanation. But currently this is how it is. It's also kind of cool.

I think the code here is related to that, making sure that if a someone expects an _interface_ that has a `handle` method and we pass in an `inflight` function of that has the same signature (looks like) the handle method, then it'll be ok (not an error). I'm not sure how this relates to the issue mentioned above. Two things:
1. @Chriscbr can we remove this hack?
2. You can try to remove this entire match arm from the match and see if the tests pass and if anything changes in the snapshots. If anything does it'll hint we still have an issue. 

Good catch noticing this issue is already closed.

Reply:
Sounds right to me 👍  - in theory it should be safe to remove this branch, since it's only really relevant for inflight closures, which are now converted into single-method classes (I don't think we had that compiler phase implemented back when this type checking rule was first added)

Reply:
Removing it seems to fails 7 tests all todo with `util.waitUntil`:
```log
hangar:test:  FAIL  src/test_corpus/sdk_tests/queue/push.main.w.test.ts > wing compile -t tf-aws
hangar:test: AssertionError:
hangar:test: error: Expected type to be "inflight (): bool", but got "inflight (): bool" instead
hangar:test:    --> /mnt/c/Users/Lance/Documents/projects/wing/examples/tests/sdk_tests/queue/push.main.w:18:25
hangar:test:    |
hangar:test: 18 |     assert(util.waitUntil((): bool => {
hangar:test:    | /-------------------------^
hangar:test: 19 | |     return q.approxSize() == 1;
hangar:test: 20 | |   }));
hangar:test:    | \---^ Expected type to be "inflight (): bool", but got "inflight (): bool" instead
```

[logs](https://github.com/Lancear/wing/blob/88eb7610b672ed85922275f895b25c60094ebe6b/failed-tests-hack.log)